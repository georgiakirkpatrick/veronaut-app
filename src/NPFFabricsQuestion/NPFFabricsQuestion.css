.NPFFabricsQuestion p {
    padding: 0;
}

.NPFFabricsQuestion .FormCheckboxSection {
    margin: 1vh 0;
}

.NPFFabricsQuestion .FormCheckboxSection p {
    padding: 1vh 0;
}

.NPFFabricsQuestion .FormCheckboxSection__option {
    padding: 1vh;
}

@media screen and (min-width: 768px) {
    .NPFFabricsQuestion .FormFieldset {
        padding: 2vh 5vw;
    }
}

@media screen and (min-width: 1024px) {
    .NPFFabricsQuestion .FormCheckboxSection p {
        padding: 0;
    }

    .NPFFabricsQuestion .FormFieldset {
        padding: 5vh 5vw;
    }
}

@media screen and (min-width: 1024px) and (min-height: 1366px) {
    .NPFFabricsQuestion .FormCheckboxSection p {
        padding: 0;
    }
}

@media screen and (min-width: 1440px) {
    .NPFFabricsQuestion .FormFieldset {
        padding: 5vh 3vw;
    }
}

@media screen and (min-width: 2560px) {
    .NPFFabricsQuestion .FormFieldset {
        padding: 3vh 3vw;
    }
}
