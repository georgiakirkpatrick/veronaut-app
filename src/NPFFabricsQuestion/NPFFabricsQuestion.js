import React from 'react'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormPage from '../FormPage/FormPage'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import NPFFooter from '../NPFFooter/NPFFooter'

const NPFFabricsQuestion = props => {
    const {
        currentPage,
        linCheck,
        primCheck,
        setBackPageTurns,
        secCheck,
        setLinBackPageTurns,
        setLinCheck,
        setPage,
        setPrimCheck,
        setPrimPageTurns,
        setSecCheck,
        setSecPageTurns
    } = props

    const linCheckOption = [
        {
            text: 'Yes',
            id: 'lining-fabric',
            name: 'lining-fabric',
            checked: linCheck
        }
    ]

    const nextButton = () => {
        if (secCheck && linCheck) {
            setPrimPageTurns(1)
            setSecPageTurns(1)
            setBackPageTurns(1)
        } else if (secCheck && !linCheck) {
            setPrimPageTurns(1)
            setSecPageTurns(2)
            setBackPageTurns(2)
        } else if (!secCheck && linCheck) {
            setLinBackPageTurns(2)
            setPrimPageTurns(2)
            setSecPageTurns(1)
            setBackPageTurns(1)
        } else if (!secCheck && !linCheck) {
            setPrimPageTurns(3)
            setBackPageTurns(3)
        }
        
        setPage(currentPage + 1)
    }

    const primCheckOption = [
        {
            text: 'Yes',
            id: 'primary-fabric',
            name: 'primary-fabric',
            checked: primCheck
        }
    ]
    
    const secCheckOption = [
        {
            text: 'Yes',
            id: 'secondary-fabric',
            name: 'secondary-fabric',
            checked: secCheck
        }
    ]


    return (
        <div id='fabric-question'>
            <FormPage title='Fabrics'>
                <FormPromptWithSub 
                    prompt='The primary fabric (or material) is the main fabric used in the garment or accessory.'
                />

                <FormCheckboxSection 
                    prompt='The primary fabric (or material) is required.'
                    options={primCheckOption}
                    selectedOptions={{'primary-fabric': primCheck}}
                    handleChange={() => setPrimCheck(primCheck)}
                />

                <FormPromptWithSub 
                    prompt='Garments and accessories sometimes include secondary fabrics as trims or accents.'
                />

                <FormCheckboxSection 
                    prompt='Does this product include a secondary fabric?'
                    options={secCheckOption}
                    selectedOptions={{'secondary-fabric': secCheck}}
                    handleChange={() => setSecCheck(!secCheck)}
                />

                <FormPromptWithSub 
                    prompt='Garments and accessories sometimes include lining fabrics - an inner layer of fabric on the interior of the product.'
                />

                <FormCheckboxSection 
                    prompt='Does this product include a lining fabric?'
                    options={linCheckOption}
                    selectedOptions={{'lining-fabric': linCheck}}
                    handleChange={() => setLinCheck(!linCheck)}
                />
            </FormPage>

            <NPFFooter 
                buttons='prevNext' 
                previousButton={() => setPage(currentPage - 1)} 
                nextButton={() => nextButton()} 
            />
        </div>
    )    
}

NPFFabricsQuestion.defaultProps = {
    currentPage: 0,
    linCheck: false,
    primCheck: false,
    secCheck: false,
    setLinCheck: () => {},
    setPage: () => {},
    setPrimCheck: () => {},
    setPrimPageTurns: () => {},
    setSecCheck: () => {},
    setSecPageTurns: () => {}
}

export default NPFFabricsQuestion