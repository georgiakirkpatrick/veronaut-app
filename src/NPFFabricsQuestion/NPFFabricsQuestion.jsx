import React from 'react'
import FormCheckboxSection from '../FormCheckboxSection/FormCheckboxSection'
import FormFieldset from '../FormFieldset/FormFieldset'
import FormPage from '../FormPage/FormPage'
import FormPromptWithSub from '../FormPromptWithSub/FormPromptWithSub'
import NPFFooter from '../NPFFooter/NPFFooter'
import './NPFFabricsQuestion.css'

const NPFFabricsQuestion = ({
  currentPage = 0,
  linCheck = false,
  primCheck = false,
  secCheck = false,
  setLinCheck = () => {},
  setPage = () => {},
  setPrimCheck = () => {},
  setPrimPageTurns = () => {},
  setSecCheck = () => {},
  setSecPageTurns = () => {}
}) => {
  const linCheckOption = [
    {
      text: 'Yes, the product has a lining fabric or material',
      id: 'lining-fabric',
      name: 'lining-fabric',
      checked: linCheck
    }
  ]

  const nextButton = () => {
    if (secCheck && linCheck) {
      setPrimPageTurns(1)
      setSecPageTurns(1)
      setBackPageTurns(1)
    } else if (secCheck && !linCheck) {
      setPrimPageTurns(1)
      setSecPageTurns(2)
      setBackPageTurns(2)
    } else if (!secCheck && linCheck) {
      setLinBackPageTurns(2)
      setPrimPageTurns(2)
      setSecPageTurns(1)
      setBackPageTurns(1)
    } else if (!secCheck && !linCheck) {
      setPrimPageTurns(3)
      setBackPageTurns(3)
    }
    
    setPage(currentPage + 1)
  }

  const primCheckOption = [
    {
      text: 'The product has a primary fabric or material',
      id: 'primary-fabric',
      name: 'primary-fabric',
      checked: primCheck
    }
  ]
  
  const secCheckOption = [
    {
      text: 'Yes, the product has a secondary fabric or material',
      id: 'secondary-fabric',
      name: 'secondary-fabric',
      checked: secCheck
    }
  ]

  return (
    <div id='fabric-question' className='NPFFabricsQuestion'>
      <FormPage title='Fabrics'>
        <FormPromptWithSub 
          prompt="Select the product's material types below"
        />

        <FormFieldset>
          {/* <FormPromptWithSub 
              prompt='The primary fabric or material is the main fabric used in the garment or accessory.'
              promptSubtitle='The primary fabric is required.'
          /> */}

          <FormCheckboxSection 
            prompt='The primary fabric or material is the main fabric used in the garment or accessory.  The primary fabric is required.'
            options={primCheckOption}
            selectedOptions={{'primary-fabric': primCheck}}
            handleChange={() => setPrimCheck(primCheck)}
          />

          {/* <FormPromptWithSub 
              promptSubtitle=''
          /> */}

          <FormCheckboxSection 
            prompt='Garments and accessories sometimes include secondary fabrics as trims or accents.  Does this product include a secondary fabric?'
            options={secCheckOption}
            selectedOptions={{'secondary-fabric': secCheck}}
            handleChange={() => setSecCheck(!secCheck)}
          />

          {/* <FormPromptWithSub 
              promptSubtitle='Garments and accessories sometimes include lining fabrics - an inner layer of fabric on the interior of the product.'
          /> */}

          <FormCheckboxSection 
            prompt='Garments and accessories sometimes include lining fabrics - an inner layer of fabric on the interior of the product.  Does this product include a lining fabric?'
            options={linCheckOption}
            selectedOptions={{'lining-fabric': linCheck}}
            handleChange={() => setLinCheck(!linCheck)}
          />
        </FormFieldset>
      </FormPage>

      <NPFFooter 
        buttons='prevNext' 
        previousButton={() => setPage(currentPage - 1)} 
        nextButton={() => nextButton()} 
      />
    </div>
  )    
}

export default NPFFabricsQuestion