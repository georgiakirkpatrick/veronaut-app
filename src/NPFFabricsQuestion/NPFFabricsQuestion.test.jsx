import React from 'react'
import { render, screen } from '@testing-library/react'
import { describe, it, expect } from 'vitest'
import NPFFabricsQuestion from './NPFFabricsQuestion'

describe('NPFFabricsQuestion', () => {
  it ('renders without crashing', () => {
    render(<NPFFabricsQuestion />)
    screen.debug()
  })

  it ('render the UI as expected', () => {
    const tree = render(<NPFFabricsQuestion />)
    expect(tree).toMatchSnapshot()
  })
})