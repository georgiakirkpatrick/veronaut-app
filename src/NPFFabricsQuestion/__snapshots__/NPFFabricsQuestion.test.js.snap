// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NPFFabricsQuestion render the UI as expected 1`] = `
<div
  className="NPFFabricsQuestion"
  id="fabric-question"
>
  <section
    className="FormPage"
  >
    <h1>
      Fabrics
    </h1>
    <form
      autoComplete="off"
      id="form-page"
    >
      <div
        className="FormPromptWithSub"
      >
        <p
          className="FormPromptWithSub__prompt"
        >
          Select the product's material types below
        </p>
        <p
          className="FormPromptWithSub__subtitle"
        >
          
        </p>
      </div>
      <fieldset
        className="FormFieldset"
      >
        <div />
        <div
          className="FormCheckboxSection"
        >
          <p>
            The primary fabric or material is the main fabric used in the garment or accessory.  The primary fabric is required.
          </p>
          <div
            className="FormCheckboxSection__options-container"
          >
            <div
              className="FormCheckboxSection__option"
            >
              <input
                checked={false}
                id="primary-fabric"
                name="primary-fabric"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="primary-fabric"
              >
                The product has a primary fabric or material
              </label>
            </div>
          </div>
        </div>
        <div
          className="FormCheckboxSection"
        >
          <p>
            Garments and accessories sometimes include secondary fabrics as trims or accents.  Does this product include a secondary fabric?
          </p>
          <div
            className="FormCheckboxSection__options-container"
          >
            <div
              className="FormCheckboxSection__option"
            >
              <input
                checked={false}
                id="secondary-fabric"
                name="secondary-fabric"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="secondary-fabric"
              >
                Yes, the product has a secondary fabric or material
              </label>
            </div>
          </div>
        </div>
        <div
          className="FormCheckboxSection"
        >
          <p>
            Garments and accessories sometimes include lining fabrics - an inner layer of fabric on the interior of the product.  Does this product include a lining fabric?
          </p>
          <div
            className="FormCheckboxSection__options-container"
          >
            <div
              className="FormCheckboxSection__option"
            >
              <input
                checked={false}
                id="lining-fabric"
                name="lining-fabric"
                onChange={[Function]}
                type="checkbox"
              />
              <label
                htmlFor="lining-fabric"
              >
                Yes, the product has a lining fabric or material
              </label>
            </div>
          </div>
        </div>
      </fieldset>
    </form>
    <div>
      
    </div>
  </section>
  <footer
    className="NPFFooter"
  >
    <button
      className="NPFFooter__previous"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-left fa-w-8 NPFFooter__left-arrow"
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        PREVIOUS
      </span>
    </button>
    <button
      className="NPFFooter__next"
      onClick={[Function]}
      type="button"
    >
      <span>
        NEXT
      </span>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right fa-w-8 NPFFooter__right-arrow"
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
  </footer>
</div>
`;
